uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/SingleWpjApiFragment.java

```diff
@@ -0,0 +1,213 @@
+package com.microsoft.identity.client.ui.automation.broker;
+
+import static com.microsoft.identity.client.ui.automation.utils.CommonUtils.FIND_UI_ELEMENT_TIMEOUT;
+
+import androidx.annotation.NonNull;
+import androidx.test.platform.app.InstrumentationRegistry;
+import androidx.test.uiautomator.UiDevice;
+import androidx.test.uiautomator.UiObject;
+import androidx.test.uiautomator.UiSelector;
+
+import com.microsoft.identity.client.ui.automation.interaction.IPromptHandler;
+import com.microsoft.identity.client.ui.automation.interaction.PromptHandlerParameters;
+import com.microsoft.identity.client.ui.automation.interaction.PromptParameter;
+import com.microsoft.identity.client.ui.automation.interaction.microsoftsts.AadPromptHandler;
+import com.microsoft.identity.client.ui.automation.interaction.microsoftsts.AdfsPromptHandler;
+import com.microsoft.identity.client.ui.automation.interaction.microsoftsts.MicrosoftStsPromptHandler;
+import com.microsoft.identity.client.ui.automation.utils.UiAutomatorUtils;
+
+import org.junit.Assert;
+
+/**
+ * A representation of the single wpj api fragment that handles all the interactions with the UI.
+ */
+public class SingleWpjApiFragment extends AbstractBrokerHost {
+    private final static String CERT_INSTALLER_PACKAGE_NAME = "com.android.certinstaller";
+    // Resource Id for the buttons
+    public final static String GET_WPJ_ACCOUNT_BUTTON_ID = "button_get_wpj_upn";
+    private final static String LEAVE_BUTTON_ID = "button_leave";
+    public final static String GET_DEVICE_STATE_BUTTON_ID = "button_get_device_state";
+    public final static String IS_DEVICE_SHARED_BUTTON_ID = "button_is_device_shared";
+    public final static String GET_DEVICE_ID_BUTTON_ID = "button_get_device_id";
+    public final static String JOIN_BUTTON_ID = "button_join";
+    public final static String JOIN_SHARED_DEVICE_BUTTON_ID = "button_join_shared_device";
+    public final static String INSTALL_CERT_BUTTON_ID = "button_install_cert";
+    public final static String JOIN_TENANT_BUTTON_ID = "button_join_tenant";
+    // Resource Id for the edit text
+    public final static String TENANT_EDIT_TEXT = "edit_text_tenant_id";
+
+    /**
+     * Perform a wpj leave operation.
+     */
+    public void wpjLeave() {
+        clickButton(LEAVE_BUTTON_ID);
+        dismissDialogBoxAndGetText();
+    }
+
+    /**
+     * Get the wpj account.
+     *
+     * @return the wpj account.
+     */
+    public String getWpjAccount() {
+        clickButton(GET_WPJ_ACCOUNT_BUTTON_ID);
+        return dismissDialogBoxAndGetText();
+    }
+
+    /**
+     * Gets the device state.
+     *
+     * @return the device state.
+     */
+    public String getDeviceState() {
+        clickButton(GET_DEVICE_STATE_BUTTON_ID);
+        return dismissDialogBoxAndGetText();
+    }
+
+    /**
+     * Checks if the device is shared.
+     *
+     * @return true if the device is shared, false otherwise.
+     */
+    public boolean isDeviceShared() {
```

Comment:
Do we have a way to access this from BrokerHost.java? Let's keep that class as the only one that devs need to access

Reply:
BrokerHost.java remains as the entry point for all operation related to the broker